docs/proposals/scheduling/20250611-networktopology-aware-scheduling.md

```diff
@@ -0,0 +1,438 @@
+| Story 2 | Node4~Node11， total 8 available nodes  | podGroup.minNumber=4( DP=2, PP=2)  <br/>priority_class = best-effort | pod-2-0:node4,  <br/>pod-2-1:node5,  <br/>pod-2-2:node6,  <br/>pod-2-3:node7 |
+| Story 3 | Node8~Node11， total 4 available nodes  | podGroup.minNumber=8( Dp=2, PP=4)  <br/>priority_class = Guarantee | pod-2-0, pod-2-1, pod-2-2, pod-2-3 was eviction。  <br/><br/>Scheduling results：<br/>pod-3-0:node4,<br/>pod-3-1:node5,<br/>pod-3-2:node6,  <br/>pod-3-3:node7,  <br/>pod-3-4:node8,  <br/>pod-3-5:node9,  <br/>pod-3-6:node10,  <br/>pod-3-7:node11 |
+
+#### 1. Management of Network Topology
+Describe the network topology through node labels and generate a configmap of the network topology within the cluster.
+
+#### 2. Definition of the Core Structure of Network Topology
+- HyperNode: is a performance domain consisting of a set of nodes or sub performance domains. Within a supernode, the network bandwidth and latency are the same. This custom resource (CRD) is used to describe the network topology in a Kubernetes cluster.
+- Tier: It is a way to distinguish between different energy domains. Within the same level, bandwidth and latency are the same. The smaller the value of the hierarchy, the higher the bandwidth. For example, computing networks and storage networks can be at different levels, or there are several levels (spine, leaf) of switches in the computing network, each level can be identified as a level.
+For example, in network architecture 1 (spine leaf), assuming 8 nodes are connected, as shown in the following figure (both single plane and multi plane can be supported):
+
+![image](/docs/images/networktopo-8-spine-leaf.png)
+
+The format of the Config Map is as follows:
+```
+[
+ {
+    "layer": 2,
+    "name": "s2",
+    "children": [
+      "s0",
+      "s1"
+    ]
+  },
+  {
+    "layer": 1,
+    "name": "s0",
+    "parents": [
+      "s2"
+    ],
+    "children": [
+      "node0",
+      "node1"
+    ]
+  },
+  {
+    "layer": 1,
+    "name": "s1",
+    "parents": [
+      "s2"
+    ],
+    "children": [
+      "node2",
+      "node3"
+    ]
+  }
+]
+
+```
+
+
+#### 3. Creation and updating of network topology
+Discovery and detection tools for network topology：
+![image](/docs/images/networktopo-9-topo-gen.png)
+
+
+#### 4. Network topology algorithm
+```
+
+// FindBestNode find N best node . （N = minNumberWorker）
+// 1.all node in the same tor
+// 2.all pipeline parallel node in the same tor, but all node in the same leaf
+// 3.all node in the same leaf
+// 4.all pipeline parallel node in the same tor, but all node in the same spine
+// 4.all pipeline parallel node in the same leaf, but all node in the same spine
+// 5.all node in the same spine
+func (nt *NetWorkTopology) FindBestNode(minNumberWorker int, pipelineParallel int, hyperNodeTree map[string][][]string) ([]string, int) {
```

Comment:
Since this func does not check the detailed spec for each pod of the same job, does it suppose that the specs of these pods are homogeneous?